src/vmm/src/guest_config/cpuid/mod.rs

```diff
@@ -317,69 +316,27 @@ pub enum CpuidTryFromKvmCpuid {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Cpuid {
     /// Intel CPUID specific information.
-    Intel(IntelCpuid),
+    Intel(BTreeMap<CpuidKey, CpuidEntry>),
     /// AMD CPUID specific information.
-    Amd(AmdCpuid),
+    Amd(BTreeMap<CpuidKey, CpuidEntry>),
```

Comment:
I think if we are removing the types it may make sense to re-structure it to look like:
```rust
pub struct Cpuid(BTreeMap<CpuidKey, CpuidEntry>);
pub enum CpuidManufacturer { Amd, Intel }
impl Cpuid {
    pub fn manufacturer(&self) -> CpuidManufacturer { /* .. */ }
}
```